packages/lesswrong/components/comments/CommentsItem/CommentsItem.tsx

```diff
@@ -311,6 +400,45 @@ export const CommentsItem = ({
 
   const voteProps = useVote(comment, "Comments", votingSystem);
   const showInlineCancel = replyFormIsOpen && isMinimalist
+  const { create: createDoppelVote, data: newDoppelCommentVote } = useCreate({
+    collectionName: "DoppelCommentVotes",
+    fragmentName: "DoppelCommentVotesFragment",
+  })
+
+  const shouldWeDoppel = userHasDoppelComments(currentUser) // is LW/AF & is admin
+    && comment.doppelComments.length >= 2 // are there doppel comments to show?
+    && !(comment.ownDoppelCommentVote || newDoppelCommentVote) // has user already voted?
+
+  const potentialCommentsData = shouldWeDoppel
+    ? { potentialComments: potentialCommentBodies,
+        setActivePotentialComment, activePotentialComment,
+        doppelCommentVote: <DoppelCommentVote
+          className=''
+          comment={comment}
+          createDoppelVote={createDoppelVote}
+          potentialCommentBodies={potentialCommentBodies}
+          activePotentialComment={activePotentialComment}
+        />
+      }
+    : undefined
+  const replaceReplyButtonsWith = potentialCommentsData ? {
+    replaceReplyButtonsWith: (comment: CommentsList) => <DoppelCommentVote
+      className={classes.replyLink}
+      comment={comment}
+      createDoppelVote={createDoppelVote}
+      potentialCommentBodies={potentialCommentBodies}
+      activePotentialComment={activePotentialComment}
+    />
+  } : {}
+
+  const makeDoppelCommentFraction = (choices: (String | null)[]): {numerator: number, denominator: number} => ({
+    numerator: choices.filter(c => c === null).length,
+    denominator: choices.length
+  }) 
+
+  const doppelCommentFraction = comment.ownDoppelCommentVote
```

Comment:
This probably wants to be wrapped in a `useMemo`, since objects which are used as props are a frequent cause of unnecessary rerenders because they often get recreated with the same value (but react components check props for reference equality)